### PR TITLE
Allow WCS merging to work without loading all data at once

### PR DESCRIPTION
**Pull request by pkgw**
For `find_optimal_celestial_wcs`, we only really need the input image shape, not the actual data. Modify the support infrastructure to allow computation in this regime. This allows us to solve for the optimal WCS even in cases where we wouldn't be able to hold all of the input images in memory at the same time.

### CODE REVIEW

**Comment by pkgw** [comment]
Numerous CI failures but they seem to be unrelated to this PR, I think. @astrofrog is this a known situation?

**Comment by Cadair** [comment]
oh my what are those test faliures! :see_no_evil:

**Comment by pkgw** [comment]
@Cadair Not a ton of activity on this repo lately ... do you know who might know how to fix up the CI?

**Comment by Cadair** [comment]
I can add it to my stack, but I also don't have a huge amount of time. @astrofrog might be able to take a look though, think he said he might be swinging back to reproject soon.

**Comment by astrofrog** [comment]
I'll take a look now

**Comment by astrofrog** [comment]
@pkgw - can you rebase? The CI should be fixed now. Thanks!

**Comment by pkgw** [comment]
Looks like I need to beef up the test coverage ...

**Comment by pkgw** [comment]
I've added some tests that should hopefully get the code coverage happy again. In so doing I did discover an implementation mistake, so yay tests!

**Comment by pkgw** [comment]
Figuring out the best approach to the Linux docs build. Meanwhile, the macOS doc build seems to be failing for an unrelated reason?

**Comment by pkgw** [comment]
OK, I think the only remaining issue is this MacOS docs thing. @astrofrog @Cadair any idea what's up here? Naively it looks like some kind of binary problem in Numpy. Presumably associated with a new release if it just started happening?

```
  File "./reproject/spherical_intersect/setup_package.py", line 3, in <module>
    import numpy as np
  File "/Users/runner/work/1/s/.tox/.package/lib/python3.9/site-packages/numpy/__init__.py", line 286, in <module>
    raise RuntimeError(msg)
RuntimeError: Polyfit sanity test emitted a warning, most likely due to using a buggy Accelerate backend. If you compiled yourself, see site.cfg.example for information. Otherwise report this to the vendor that provided NumPy.
RankWarning: Polyfit may be poorly conditioned
```

**Comment by pkgw** [comment]
Ah: https://github.com/numpy/numpy/issues/15947

**Comment by pkgw** [comment]
Finally green! Let me know if you'd like to see anything else before merging.

Once this is merged, would you mind putting out a new release with the new API feature? I'd like to be able to start using this in [toasty](https://github.com/WorldWideTelescope/toasty/).

File: reproject/mosaicking/wcs_helpers.py [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
                                skycoord_to_pixel, wcs_to_celestial_frame)
 
-from ..utils import parse_input_data
+from ..utils import parse_input_shape
 
 __all__ = ['find_optimal_celestial_wcs']
 
 
-def find_optimal_celestial_wcs(input_data, frame=None, auto_rotate=False,
+def find_optimal_celestial_wcs(input_shapes, frame=None, auto_rotate=False,
```
Comment (astrofrog) [inline reproject/mosaicking/wcs_helpers.py:14]: I'm hesitant to change the argument name as I've seen cases where the first arg is passed as a kwarg - can you change the name back?

**Review by astrofrog — CHANGES_REQUESTED** [review]
Looks good! Just one comment below.

**Comment by astrofrog** [comment]
This also needs a changelog entry

**Comment by pkgw** [comment]
OK, this should do it ...

**Review by astrofrog — APPROVED** [review]
Looks good, thanks!

**Pull request merged**